### ISSUE TITLE
install dependencies needed to build

### ISSUE THREAD

**Issue by cboettig** [traitecoevo/drat#1]
@richfitz really enjoying your scripts here, which are rather nicer than what I have in ropensci/packages.  It would be nice if the scripts attempted to install the dependencies if they were not available.  This wouldn't be fail-safe of course (e.g. with any non-CRAN hosted dependencies, but of course install_github doesn't handle that either), but it would facilitate those cases where you're installing from the Github page of some third-party package that decides to tack on an additional dependency that isn't available in the local build system.  

I was just calling `install_github` before attempting to build the package for this reason, but clearly that's an inefficient hack since it pulls down the repos again.  Curious if you had a better solution in mind that would accomplish the same thing.

**Comment by richfitz** [comment]
Do packages need other packages to build?  I guess that makes sense for doing the vignettes, right?  Is it needed for any non-vignette packages

The first step is to download all the source packages; we could install locally from there without hitting the  github versions again.  Probably best to do what `R CMD check` does and install into a temporary directory so as not to change system installed packages (though I assume you're doing this in a container).

I guess the trick is that some packages need things in their Suggests but it's hard to know which of those are going to be used...

**Comment by cboettig** [comment]
Yup, I guess technically this requires having the full depends/imports/suggests list.  

I suppose alternatively one could build the packages without building the vignettes if the full dependencies aren't available.

**Comment by cboettig** [comment]
Just hacked together a function to install only missing packages rather than calling `install_github`, see 
https://github.com/cboettig/drat/commit/b51ae858cc385783a41ad52c2109557a5542fba3

Currently just installs the depends and imports but would be easy to modify.
